docs/tutorial/resift-rentals.md

```diff
@@ -0,0 +1,2110 @@
+
+#### Show Fetch Status
+
+We want to show a loading spinner when we're fetching the data. To achieve this, we'll use the `useStatus` and a helper function to check is the status is 'loading' called `isLoading`. And we'll grab the Material UI spinner called `CircularProgress`. Let's get them imported.
+
+```js
+import { useStatus, isLoading } from 'resift';
+import { CircularProgress } from '@material-ui/core';
+```
+
+Now we can add the spinner in while loading
+
+```js
+function App() {
+  ...
+  const status = useStatus(getGenres)
+
+  return (
+    <>
+      ...
+      {isLoading(status) && <CircularProgress className={classes.spinner} />}
+      ...
+    </>
+  );
+}
+```
+
+Our complete file looks like this now:
+
+```js
+// App.js
+import React, { useEffect } from 'react';
+// Components
+import AppBar from 'components/AppBar';
+import Genre from 'components/Genre';
+// Fetches
+import { useData, useDispatch, Guard, useStatus, isLoading } from 'resift';
+import getGenres from 'fetches/getGenres';
+// Styles
+import { makeStyles } from '@material-ui/core/styles';
+import { CircularProgress } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  root: {},
+  genre: {
+    margin: '24px 0',
+  },
+  spinner: {
+    color: 'white',
+  },
+}));
+
+function App() {
+  const classes = useStyles();
+  const genres = useData(getGenres);
+  const dispatch = useDispatch();
+  const status = useStatus(getGenres);
+
+  useEffect(() => {
+    dispatch(getGenres());
+  }, [dispatch]);
+
+  return (
+    <div className={classes.root}>
+      <AppBar />
+      {isLoading(status) && <CircularProgress className={classes.spinner} />}
+      <Guard fetch={getGenres}>
+        {genres =>
+          genres.map(genre => <Genre key={genre.id} genre={genre} className={classes.genre} />)
+        }
+      </Guard>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Refresh the page and you shall see the loading spinner before the genres data kick in.
+
+### Conclude
+
+Now you’ve gone through some basic fetch concepts, let's review the online order analogy we made earlier to help form a sticky mental modal.
+
+- _Creating a fetch factory is like adding what you want in a cart._</br>
+  You're defining what your order/fetch looks like.
+- _Creating the fetch instance is like submitting your order._</br>
+  You've made up your mind and confirmed your order/fetch.
+- _Using the fetch is like sending your order to the fulfillment facility, for which the fulfillment facility will respond with the goods you ordered, and send you your fulfillment status._</br>
+  You send off your order/fetch, and then the fulfillment facility/server respond with the goods/data you requested, and notify you the status of sending over the goods/data.
+- _Dispatching the fetch is the fulfillment facility sending things out to you based on your order._</br>
+  The server sending out the data requested.
```

Comment:
same comment as above. I'm not sure this analogy makes sense. We can work on this or even simply omit it for this tutorial if you agree.

Reply:
updated in my next commit